ops/pebble.py

```diff
@@ -701,6 +727,78 @@ def __repr__(self):
                 ).format(self=self)
 
 
+class Check:
+    """Represents a check configuration in a Pebble configuration layer."""
+
+    def __init__(self, name: str, raw: typing.Dict = None):
```

Comment:
Why the choice to make raw an untyped dict instead of unpacking it?
I would rather have `Check('name', override:str='', level:str='', period:str='', ...)`, makes for a way easier to read initializer, and easier usage of the class.
If you really dislike doing `check=Check('name', **my_raw_dict)`, then we should expose an alternative classmethod
`Check.from_dict(name:str, dict_:dict)`.

If we were on more modern python I could also go for a dataclass or a TypedDict, but until then, this feels like better code.

Reply:
Actually the whole Check class could very well be a dataclass, and save some __eq__, __repr__ and to_dict boilerplate

Reply:
Yeah, this should arguably be done with a `from_dict`. However, I've made this consistent with the other layer-configuration classes, which all do it the same way. If doing it again maybe we could have considered dataclasses or some such (then again, that's 3.7+ and `ops` still needs to support 3.5 for now). So I'm going to leave as is to be consistent.